### DISCUSSION TITLE
Error: SHA256 mismatch when installing android-platform-tools r34.0.4

### DISCUSSION THREAD

**Discussion by aircraft-cerier** [Homebrew/discussions#4661] [Everyday usage]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.1.0
ORIGIN: https://github.com/Homebrew/brew
HEAD: ac93842f8d13913f462aedb246cfacd08d6b7f05
Last commit: 2 days ago
Core tap JSON: 21 Jul 21:09 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.39.2 => /Applications/Xcode15.0beta4.app/Contents/Developer/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-arm64
CLT: 15.0.0.0.1.1688827124
Xcode: 15.0 => /Applications/Xcode15.0beta4.app/Contents/Developer
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

Hi,

I'm running:
`brew install android-platform-tools
`
The output is:

```
==> Downloading https://formulae.brew.sh/api/formula.jws.json

==> Downloading https://dl.google.com/android/repository/platform-tools_r34.0.4-darwin.zip
############################################################################################### 100.0%
Error: SHA256 mismatch
Expected: bd09b834c150181a383deba63e157e9a53a6eb5a9cf4849b2b79dd89d0a2ddf1
  Actual: c60d9ee0acade673b649543b2d53c632ba54f9ca083ed612176105117fa16102
    File: /Users/brcerier/Library/Caches/Homebrew/downloads/9b8e257fff105a69876e9b1fd1ce420143d0bf4001e48cdbde5fbd9d848bdad9--platform-tools_r34.0.4-darwin.zip
To retry an incomplete download, remove the file above.

```
Also get similar error with this command after running the remove command on the file:

`brew install --cask android-platform-tools
`

Output:
```
==> Downloading https://dl.google.com/android/repository/platform-tools_r34.0.4-darwin.zip
############################################################################################### 100.0%
Error: SHA256 mismatch
Expected: bd09b834c150181a383deba63e157e9a53a6eb5a9cf4849b2b79dd89d0a2ddf1
  Actual: c60d9ee0acade673b649543b2d53c632ba54f9ca083ed612176105117fa16102
    File: /Users/brcerier/Library/Caches/Homebrew/downloads/9b8e257fff105a69876e9b1fd1ce420143d0bf4001e48cdbde5fbd9d848bdad9--platform-tools_r34.0.4-darwin.zip
To retry an incomplete download, remove the file above.

```
Does anyone have ideas on how to resolve this checksum error?
Seems like a similar issue to: https://github.com/orgs/Homebrew/discussions/4501

Thank you.

**Comment by osalbahr** [comment]
I can confirm that the same happens on my end:

```console
$ brew doctor   
Your system is ready to brew.
$ brew install android-platform-tools
==> Downloading https://dl.google.com/android/repository/platform-tools_r34.0.4-
######################################################################### 100.0%
Error: SHA256 mismatch
Expected: bd09b834c150181a383deba63e157e9a53a6eb5a9cf4849b2b79dd89d0a2ddf1
  Actual: c60d9ee0acade673b649543b2d53c632ba54f9ca083ed612176105117fa16102
    File: /Users/dracula/Library/Caches/Homebrew/downloads/9b8e257fff105a69876e9b1fd1ce420143d0bf4001e48cdbde5fbd9d848bdad9--platform-tools_r34.0.4-darwin.zip
To retry an incomplete download, remove the file above.
$ brew doctor
Your system is ready to brew.
$ rm /Users/dracula/Library/Caches/Homebrew/downloads/9b8e257fff105a69876e9b1fd1ce420143d0bf4001e48cdbde5fbd9d848bdad9--platform-tools_r34.0.4-darwin.zip
$ brew install --cask android-platform-tools
==> Downloading https://dl.google.com/android/repository/platform-tools_r34.0.4-
######################################################################### 100.0%
Error: SHA256 mismatch
Expected: bd09b834c150181a383deba63e157e9a53a6eb5a9cf4849b2b79dd89d0a2ddf1
  Actual: c60d9ee0acade673b649543b2d53c632ba54f9ca083ed612176105117fa16102
    File: /Users/dracula/Library/Caches/Homebrew/downloads/9b8e257fff105a69876e9b1fd1ce420143d0bf4001e48cdbde5fbd9d848bdad9--platform-tools_r34.0.4-darwin.zip
To retry an incomplete download, remove the file above.
$ brew config
HOMEBREW_VERSION: 4.1.0-15-g5226cd4
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5226cd434494991fae9da7eb6ab6a3095a16f563
Last commit: 3 hours ago
Core tap JSON: 21 Jul 23:26 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 4
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: quad-core 64-bit kabylake
Clang: 14.0.3 build 1403
Git: 2.39.2 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 7.88.1 => /usr/bin/curl
macOS: 13.4.1-x86_64
CLT: 14.3.1.0.1.1683849156
Xcode: N/A
```

---

> Does anyone have ideas on how to resolve this checksum error?

This needs a PR to fix the checksum, perhaps this will PR will fix it:

- Homebrew/homebrew-cask#151454

For more: [Cask - checksum does not match](https://docs.brew.sh/Common-Issues#cask---checksum-does-not-match)

**Comment by chenrui333** [comment]
fixed by https://github.com/Homebrew/homebrew-cask/pull/151454

**Comment by GuilioM** [comment]
Hi, 

Did this problem return? Here's my config and error after running 'brew install android-platform-tools` and `brew install --cask android-platform-tools` on MacBook Air M1 Sequoia 15.5:

`HOMEBREW_VERSION: 4.5.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: c19f08b3e8ea46ba94bd9315c46c0dda10ead31c
Last commit: 10 hours ago
Branch: stable
Core tap JSON: 23 Jun 23:01 UTC
Core cask tap JSON: 23 Jun 23:01 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.4.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.4.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 17.0.0 build 1700
Git: 2.39.5 => /Library/Developer/CommandLineTools/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.5-arm64
CLT: 16.4.0.0.1.1747106510
Xcode: N/A
Rosetta 2: false
`

`==> Downloading https://dl.google.com/android/repository/platform-tools_r36.0.0-darwin.zip
################################################################################################################################################################################# 100.0%
Error: SHA256 mismatch
Expected: b241878e6ec20650b041bf715ea05f7d5dc73bd24529464bd9cf68946e3132bd
  Actual: d3e9fa1df3345cf728586908426615a60863d2632f73f1ce14f0f1349ef000fd
    File: /Users/XXX/Library/Caches/Homebrew/downloads/f3c9b639b8124f894ed86a6fdf9111692e4fef9226365692e60ab6749b01a735--platform-tools_r36.0.0-darwin.zip
To retry an incomplete download, remove the file above.`

**Comment by sonysasankan** [comment]
Yup, Im getting the same SHA256 mismatch error

**Comment by sno-man** [comment]
same on my end on Apple M1 Max Ventura 13.7.6

```
==> Downloading https://dl.google.com/android/repository/platform-tools_r36.0.0-darwin.zip
Already downloaded: /Users/som/Library/Caches/Homebrew/downloads/f3c9b639b8124f894ed86a6fdf9111692e4fef9226365692e60ab6749b01a735--platform-tools_r36.0.0-darwin.zip
Error: SHA256 mismatch
Expected: b241878e6ec20650b041bf715ea05f7d5dc73bd24529464bd9cf68946e3132bd
  Actual: d3e9fa1df3345cf728586908426615a60863d2632f73f1ce14f0f1349ef000fd
    File: /Users/som/Library/Caches/Homebrew/downloads/f3c9b639b8124f894ed86a6fdf9111692e4fef9226365692e60ab6749b01a735--platform-tools_r36.0.0-darwin.zip
To retry an incomplete download, remove the file above.
```

**Comment by p-linnane** [comment]
The reason you're all seeing that is that upstream replaced the file without incrementing the version number. The expected hash was working at one time. This is a poor practice on Google's part. Please open a PR to update the hash.

**Comment by 4jie** [comment]
I meet the same problem on mac m4pro, too.

**Reply by osalbahr** [reply]
Try again. I just installed and it went fine on M2 Pro. If the problem still persists, you might want to paste the command output.

**Reply by 4jie** [reply]
I'm sorry i'm just seeing this now,but i'm try just now had the same problem.I use the command "brew install android-platform-tools" and "brew install --cask android-platform-tools".
